src/services/pncWebSocketService.ts

```diff
@@ -0,0 +1,54 @@
+import { pncWebSocketClient } from 'services/pncWebSocketClient';
+import { uiLogger } from 'services/uiLogger';
+
+type TWebSocketListener = (messageEvent: MessageEvent) => void;
+const messageListeners: Array<TWebSocketListener> = [];
```

Comment:
Global state pollution. This is poorly extendable.
Generic websocket class having this state (and also listener add/remove methods) would be preferable.
Then, `pncWebSocketClient` would be an instance of that class.